src/v/raft/recovery_stm.cc

```diff
@@ -429,7 +430,8 @@ ss::future<> recovery_stm::replicate(
         .term = _term,
         .prev_log_index = prev_log_idx,
         .prev_log_term = prev_log_term,
-        .last_visible_index = last_visible_idx},
+        .last_visible_index = last_visible_idx,
+        .dirty_offset = _ptr->dirty_offset()},
```

Comment:
nit: we query log offsets few lines below maybe we can move the query before we need dirty offset here so we will skip one `log::offsets()` call